modules/spring-data-ext/src/main/java/org/apache/ignite/springdata/repository/support/IgniteResourceProviderImpl.java

```diff
@@ -0,0 +1,135 @@
+ * the License.  You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.ignite.springdata.repository.support;
+
+import org.apache.ignite.Ignite;
+import org.apache.ignite.Ignition;
+import org.apache.ignite.client.IgniteClient;
+import org.apache.ignite.configuration.ClientConfiguration;
+import org.apache.ignite.configuration.IgniteConfiguration;
+import org.apache.ignite.springdata.proxy.ClosableIgniteClientProxy;
+import org.apache.ignite.springdata.proxy.ClosableIgniteProxy;
+import org.apache.ignite.springdata.proxy.IgniteClientProxy;
+import org.apache.ignite.springdata.proxy.IgniteProxy;
+import org.apache.ignite.springdata.proxy.IgniteProxyImpl;
+import org.apache.ignite.springdata.repository.config.RepositoryConfig;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.DisposableBean;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+
+/** Represents default implementation of {@link IgniteResourceProvider} */
+public class IgniteResourceProviderImpl implements IgniteResourceProvider, ApplicationContextAware, DisposableBean {
+    /** Spring application context. */
+    @SuppressWarnings("FieldAccessedSynchronizedAndUnsynchronized")
+    private ApplicationContext ctx;
+
+    /** Ignite proxy. */
+    private volatile IgniteProxy igniteProxy;
+
+    /** {@inheritDoc} */
+    @Override public IgniteProxy igniteProxy() {
+        IgniteProxy res = igniteProxy;
+
+        if (res == null) {
+            synchronized (this) {
+                res = igniteProxy;
+
+                if (res == null) {
+                    res = createIgniteProxy();
+
+                    igniteProxy = res;
+                }
+            }
+        }
+
+        return igniteProxy;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void setApplicationContext(ApplicationContext ctx) throws BeansException {
+        this.ctx = ctx;
+    }
+
+    /** {@inheritDoc} */
+    @Override public void destroy() throws Exception {
+        IgniteProxy proxy = igniteProxy;
+
+        if (proxy instanceof AutoCloseable)
+            ((AutoCloseable)proxy).close();
+    }
+
+    /**
+     * Creates {@link IgniteProxy} to be used for providing access to the Ignite cluster for all repositories.
+     *
+     * @return Instance of {@link IgniteProxy}.
+     *
+     * @see RepositoryConfig
+     */
+    private IgniteProxy createIgniteProxy() {
+        try {
+            Object igniteInstanceBean = ctx.getBean("igniteInstance");
```

Comment:
User may wants to run multiple instances of a client / server or both of them. Looks like it is a valid case, but ctx.getBean does not handle it, isn't it?

Reply:
I keep the existing logic unchanged just moved it into a separate bean. Currently, configuration of multiple Ignite instances is not supported for this version of Spring Data integration.